piptools/sync.py

```diff
@@ -2,7 +2,7 @@
 import os
 import sys
 import tempfile
-from subprocess import check_call  # nosec
```

Comment:
> many useless "nosec" comments for each subrocess call

There must be an option to add `check_call` to `allow-list`.

Reply:
I think the comments must remain where they are: the check is not about whether the import is dangerous, it's rather about how the function is called and if any user input is passed there unsanitized. Such things must be checked on a case-by-case basis. You cannot just globally declare it safe and hope for the better.
The only case when it's fine to ignore it is tests: https://github.com/ansible/pylibssh/blob/afabcea/.flake8#L89.
By the way, I like [wemake-python-styleguide](https://wemake-python-stylegui.de) which is a collection of flake8 linters pulled in together. It uses flake8-bandit that is quite handy because you get to use flake8 ignore machinery when necessary and it becomes a part of the linting suite where it belongs. I recommend switching to it.

Reply:
> I think the comments must remain where they are: the check is not about whether the import is dangerous, it's rather about how the function is called and if any user input is passed there unsanitized. Such things must be checked on a case-by-case basis. You cannot just globally declare it safe and hope for the better.

I agree that subprocess has the potential for misue, but that isn't the case with pip-tools. These comments just add noise and don't fulfill the implication of a real security audit.

Reply:
Comments could have extended explanation, yes. But the presence of them adds more context and information that actual humans checked this. If you do global ignores, there's no trace of whether the simplest problems are even checked.

Reply:
When I read these comments, I don't think these comments communicate anything useful. They just come across as terse noise. Expanding the comment seems entirely unnecessary to me. The use of subprocess looks pretty straightforward, logical, and obvious to me.

Reply:
It's not the use of subprocess that needs to be expanded but the reasoning behind the ignore comments.